### PR TITLE
Fix unreliable link test

### LINKED ISSUE #13044: Tests: Enable a disabled fragile link e2e test
In #13043 `allows autocomplete suggestions to be selected with the mouse` test from links spec was disabled because it was failing too often on Travis. 

When it fails, this is what is printed on the console:

> FAIL test/e2e/specs/links.test.js (57.317s)
>   ● Links › allows autocomplete suggestions to be selected with the mouse
>     expect(received).not.toBeNull()
>     Received: null
>       300 | 		// Expect that clicking on the autocomplete suggestion doesn't dismiss the link popover.
>       301 | 		await firstSuggestion.click();
>     > 302 | 		expect( await page.$( '.editor-url-popover' ) ).not.toBeNull();
>           | 		                                                    ^
>       303 | 
>       304 | 		// Expect the url input value to have been updated with the post url.
>       305 | 		const inputValue = await page.evaluate( () => document.querySelector( '.editor-url-input input[aria-label="URL"]' ).value );
>       at Object.toBeNull (test/e2e/specs/links.test.js:302:55)
>       at tryCatch (node_modules/regenerator-runtime/runtime.js:62:40)
>       at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:288:22)
>       at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:114:21)
>       at asyncGeneratorStep (test/e2e/specs/links.test.js:15:103)
>       at _next (test/e2e/specs/links.test.js:17:194)

In my tests, it didn't regress. It would be nice to discover why this test doesn't work as intended with Puppeteer.

/cc @talldan

### PR DESCRIPTION
**Pull request by talldan**
Attempts to fix #13044. This failure only happens in CI (unable to reproduce locally) hence this PR

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by talldan** [comment]
Hey @gziolo - I haven't really done any more to fix this one beyond a few changes to the test code. I wasn't able to work out why it was failing.

However, over the last 1-2 weeks I've been running the tests on travis in this branch on a daily basis, and I haven't seen a single failure of the test in question. I wonder if other code changes have resulted in this test becoming more stable 🤔 

I think it might be worth trying to reintroduce the test— we can always disable it again if we see some failures.

**Review by gziolo — APPROVED** [review]
Let's reintroduce this test then :)

**Comment by youknowriad** [comment]
This is still failing in several PRs for me. Let's disable all the unstable tests we find.

File: packages/e2e-tests/specs/links.test.js [hunk 1]
```diff
@@ -267,7 +267,7 @@ describe( 'Links', () => {
 
 	// Test for regressions of https://github.com/WordPress/gutenberg/issues/10496.
 	// Disabled until improved as it wasn't reliable enough.
```
Comment (aduth) [inline packages/e2e-tests/specs/links.test.js]: This comment should have been removed.

**Comment by aduth** [comment]
> This is still failing in several PRs for me. Let's disable all the unstable tests we find.

See #15059

**Pull request merged**